### PR TITLE
DAO Legacy

### PR DESCRIPTION
**Pull request by Da-Colon**
## Overview

Adds hook and UI to show a user DAO connections

## Changes
- Branch has been rebased on top of #298 
- Extracted UI components for reuse
- Update Favorites UI and functionality to match Details Page.

## Issues
There are a few issues with loading and component leaks.

- When navigating from Fractal DAO 'home' to details page. a component leak error is shown. There is a second render after load that needs be looked into not sure what is causing it?

### Details
image

### Favorites
image

### CODE REVIEW

File: src/contexts/daoData/useDAOLegacy.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      // check if creator is a Fractal DAO
+      const creatorEvent = await getPastEvents(
+        daoFactoryContract,
+        daoFactoryContract.filters.DAOCreated(creator)
+      );
+      setParentDAO(creatorEvent.length ? creator : undefined);
+      const subsidiaries = await getPastEvents(
+        daoFactoryContract,
+        daoFactoryContract.filters.DAOCreated(null, null, null, creator)
+      );
```
Comment (Da-Colon) [inline src/contexts/daoData/useDAOLegacy.ts:42]: currently this is throwing an error as the creator isn't currently indexed when the event is emitted in the contracts

Comment (Da-Colon) [reply src/contexts/daoData/useDAOLegacy.ts:41]: @tbwebb22 Can this be updated in the contracts?

**Comment by Da-Colon** [comment]
@cmccand @tbwebb22 This PR is ready for review. I have tested and listener works.

One Issue to note when navigating to an another DAO, The Proposal of the previous DAO shows and takes a second to reload (I have seen it not reload proposals) I think the problem there is bigger than this PR and involves how the app loads in a DAO

**Comment by adamgall** [comment]
@Da-Colon can you please open an issue for that proposal display bug upon new DAO navigation?

Also, I think you tagged the wrong Chris in your previous comment

**Comment by Da-Colon** [comment]
@adamgall will do, fixed the @ good catch

File: src/contexts/daoData/useDAOLegacy.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    (async () => {
+      const creationEvent = await getPastEvents(
+        daoFactoryContract,
+        daoFactoryContract.filters.DAOCreated(daoAddress)
+      );
+
+      const creator = creationEvent[0].args[3];
+
+      // checks for DAO creation event for creator address
+      const isDaoParent = (
```
Comment (christopherdancy) [inline src/contexts/daoData/useDAOLegacy.ts:41]: This is really good code - I love how simple it is!

File: src/pages/DaoFavorites/index.tsx [hunk 3]
```diff
@@ -1,52 +1,9 @@
...
 import ContentBox from '../../components/ui/ContentBox';
-import { TextButton } from '../../components/ui/forms/Button';
 import H1 from '../../components/ui/H1';
-import useDAOContract from '../../contexts/daoData/useDAOContract';
-import useDAOName from '../../contexts/daoData/useDAOName';
 import { useWeb3Provider } from '../../contexts/web3Data/hooks/useWeb3Provider';
 import useFavorites from '../../hooks/useFavorites';
-import useIsDAO from '../../hooks/useIsDAO';
-
-function FavoriteRow({ address }: { address: string }) {
```
Comment (christopherdancy) [inline src/pages/DaoFavorites/index.tsx:12]: Hey @Da-Colon why did we get rid of fav Ro.w

Comment (christopherdancy) [reply src/pages/DaoFavorites/index.tsx:12]: The details page and the favorites page used similar UI and this was confusing. David updated this to share a component between the pages.

**Review by christopherdancy — APPROVED** [review]
LGTM

**Review by tbwebb22 — APPROVED** [review]

**Pull request merged**